jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithPotOwnerQueryMapper.java

```diff
@@ -0,0 +1,118 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.kie.services.impl.query.mapper;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.dashbuilder.dataset.DataSet;
+import org.jbpm.services.api.model.UserTaskInstanceWithPotOwnerDesc;
+import org.jbpm.services.api.query.QueryResultMapper;
+
+public class UserTaskInstanceWithPotOwnerQueryMapper extends AbstractQueryMapper<UserTaskInstanceWithPotOwnerDesc> implements QueryResultMapper<List<UserTaskInstanceWithPotOwnerDesc>> {
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Dedicated for ServiceLoader to create instance, use <code>get()</code> method instead 
+     */
+    public UserTaskInstanceWithPotOwnerQueryMapper() {
+        super();
+    }
+    
+    public static UserTaskInstanceWithPotOwnerQueryMapper get() {
+        return new UserTaskInstanceWithPotOwnerQueryMapper();
+    }
+    
+    @Override
+    public List<UserTaskInstanceWithPotOwnerDesc> map(Object result) {
+        if (result instanceof DataSet) {
+            DataSet dataSetResult = (DataSet) result;
+            List<UserTaskInstanceWithPotOwnerDesc> mappedResult = new ArrayList<UserTaskInstanceWithPotOwnerDesc>();
+            
+                                                                
+            if (dataSetResult != null) {
+                Map<Long, UserTaskInstanceWithPotOwnerDesc> tmp = new HashMap<Long, UserTaskInstanceWithPotOwnerDesc>();
+                
+                for (int i = 0; i < dataSetResult.getRowCount(); i++) {
+                    Long taskId = getColumnLongValue(dataSetResult, COLUMN_TASKID, i);
+                    UserTaskInstanceWithPotOwnerDesc ut = tmp.get(taskId);
+                    if (ut == null) {
+
+                        ut = buildInstance(dataSetResult, i);
+                        mappedResult.add(ut);    
+                        
+                        tmp.put(taskId, ut);
+                    }else if(getColumnStringValue(dataSetResult, COLUMN_POTOWNER, i) != null){
+                        ut.addPotOwner(getColumnStringValue(dataSetResult, COLUMN_POTOWNER, i));
+                    }  
+                                    
+                }
+            }
+            
+            return mappedResult;
+        }
+        
+        throw new IllegalArgumentException("Unsupported result for mapping " + result);
+    }
+
+    @Override
+    public String getName() {
+        return "CustomUserTasks";
```

Comment:
it's not really CustomUserTasks but rather UserTasksWithPotOwners

Reply:
Ok!